### PR TITLE
More hpc backports

### PR DESCRIPTION
**Pull request by ChrisJefferson**
more functions backported from hpcgap

### CODE REVIEW

**Comment by alex-konovalov** [comment]
@markuspf was this tested with Travis? It seems not...

**Comment by markuspf** [comment]
No it was not, but travis has problems at the moment so we decided to not look at this before travis' problems have been resolved.
Of course github's failure to give error messages is not really helpful here either...

**Comment by ChrisJefferson** [comment]
This patch is currently in a state of flux, and should not be merged (yet!) It is here so people can observe it.

File: lib/thread1.g [hunk 1]
```diff
@@ -6,15 +6,86 @@
...
+
+# Now we mock functions from thread1.g
+
+MakeThreadLocal := function(l) return l; end;
+MakeReadOnly := function(l) return l; end;
+MakeReadOnlyRaw := function(l) return l; end;
+MakeProtected := function(l) return l; end;
+MakeProjectedObj := function(l) return l; end;
+MakeReadOnlyObj := function(l) return l; end;
+
```
Comment (fingolfin) [inline lib/thread1.g]: In HPC-GAP, these and others all are read-only, e.g.:
```
IsReadOnlyGVar("MakeThreadLocal");
```
So perhaps this should be using `BindGlobal`. I'd also consider using `IdFunc` for the value, as (a) some code optimizes for that, and (b) it saves a tiny amount of memory ;-). So we'd write:
```
BindGlobal("MakeThreadLocal", IdFunc);
```

File: lib/thread1.g [hunk 2]
```diff
@@ -6,15 +6,86 @@
...
+IsReadOnly := function(l) return false; end;
+IsProtected := function(l) return false; end;
```
Comment (fingolfin) [inline lib/thread1.g]: These two could potentially use `ReturnFalse`

**Comment by fingolfin** [comment]
My understanding, based on a discusson with @ChrisJefferson in Trondheim, is that he wants to convert this PR to use BindGlobal. After that, I think it can be merged. Of course we could also wait for #252, but I'd rather not delay this PR too much. A simple solution would be thus to just add `ReturnFirst` and `ReturnNothing` as pure GAP functions in, say, thread1.g (or whatever works), perhaps even as part of this PR here. Then we can clean it up later. (AFAICT, nobody seems to have objections to ReturnFirst and ReturnNothing so far).

**Comment by alex-konovalov** [comment]
I support introducing `ReturnFirst` and `ReturnNothing` - they could be useful for prototyping. Don't mind if they will just go to `thread1.g` first.

**Comment by ChrisJefferson** [comment]
Assuming these tests pass, I believe we can now merge #252, then this.

**Comment by alex-konovalov** [comment]
OK, I will merge 252 now.

**Comment by alex-konovalov** [comment]
Merged this as well - will be picked up by Jenkins tonight for more extensive checks.

**Comment by alex-konovalov** [comment]
@ChrisJefferson Please fix manual examples: the one for `ReturnNothing` actually uses `ReturnFail`. The other for `ReturnFirst` should be matched to a new behaviour.
```
########> Diff in [ "./../../lib/function.g", 482, 487 ]
# Input is:
n();
# Expected output:
# But found:
fail
########
########> Diff in [ "./../../lib/function.g", 482, 487 ]
# Input is:
n(-42);
# Expected output:
# But found:
fail
########
########> Diff in [ "./../../lib/function.g", 507, 516 ]
# Input is:
f:=ReturnFirst;  
# Expected output:
function( arg ) ... end
# But found:
function( object ) ... end
########
```

**Comment by ChrisJefferson** [comment]
Woops, forgot that just running 'make testmanuals' won't tell me when there is a failure.

Now fixed and pushed to master.

**Comment by alex-konovalov** [comment]
Thanks. Well, you need to look to three log files in `dev/log` and the one with `A` (`testmanualsA-timestamp`) stands for packages loaded by default and should have zero diffs.

**Pull request merged**